code-studio/src/js/code-studio.js

```diff
@@ -9,6 +9,9 @@
 'use strict';
 /* global Sounds */
 
+// Shim window.console to be safe in IE
+require('./consoleShim')(window);
```

Comment:
Could also combine B and C `window.console = require('./consoleShim')(window.console)`, where the exported object calls the underlying console function if there is one. I'm also fine with approach B - we are shimming a global native namespace, so there's going to be some amount of awkwardnesss.